compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/validator/FunctionValidator.java

```diff
@@ -400,6 +432,8 @@ private void validateReturnTypeUnion(List<TypeSymbol> returnTypeMembers, Locatio
                     if ((((TypeReferenceTypeSymbol) returnType).definition()).kind() == SymbolKind.TYPE_DEFINITION) {
                         if (!isRecordType(returnType)) {
                             updateContext(context, CompilationError.INVALID_RETURN_TYPE, location);
+                        } else if (isFileUploadParameter(returnType)) {
+                            updateContext(context, CompilationError.INVALID_RETURN_TYPE_FILE_UPLOAD, location);
```

Comment:
Do we need this check? If this is `FileUpload` record, the `stream` field should throw an error.